packages/medusa/src/services/product-variant-inventory.ts

```diff
@@ -342,20 +375,133 @@ class ProductVariantInventoryService extends TransactionBaseService {
         return await this.inventoryService_.createReservationItem({
           ...toReserve,
           location_id: locationId as string,
-          item_id: inventoryPart.inventory_item_id,
+          inventory_item_id: inventoryPart.inventory_item_id,
           quantity: itemQuantity,
         })
       })
     )
   }
 
   /**
-   * Remove reservation of variant quantity
+   * Adjusts the quantity of reservations for a line item by a given amount.
+   * @param {string} lineItemId - The ID of the line item
+   * @param {string} variantId - The ID of the variant
+   * @param {string} locationId - The ID of the location to prefer adjusting quantities at
+   * @param {number} quantity - The amount to adjust the quantity by
+   */
+  async adjustReservationsQuantityByLineItem(
+    lineItemId: string,
+    variantId: string,
+    locationId: string,
+    quantity: number
+  ): Promise<void> {
+    const manager = this.transactionManager_ || this.manager_
+
+    if (!this.inventoryService_) {
+      const variantServiceTx =
```

Comment:
suggestion(non blocking): I would move that to a private method and call it here. something like `adjustReservationsQuantityByLineItemWhithoutInventory`